### DISCUSSION TITLE
Запасной план

### DISCUSSION THREAD

**Discussion by Mintedan** [cawa-93/anime-library#52] [Новые идеи]
Все время слышу что аниме можно смотреть через торент (сам такого никогда не пробывал).
Если недай бог смотреть аниме заблокируют(или они сами закроются) может ты сделаешь какую нибудь интыграцию с этой темой или такое не возможно? Я конечно не программист, но по идее же можно сделать прогу, которая ищет на нескольких сайтах по названию аниме его торент, а потом открывает его в плеере ну например в плеере уторента.

**Comment by cawa-93** [comment]
Технически, это возможно. Когда-то я видел видео-плееры которые работают по BitTorrent протоколу. Но это, как по мне, весьма не надёжно и очень сложно реализуемо по ряду причин:
- Сама идея в основе (peer-to-peer сеть) подразумевает, что где-то должны быть люди, у которых будут нужные видео файлы. То есть, вот лежит у меня на компе серия аниме -- её могут смотреть все желающие. Я ещё удалил -- и она безвозвратно пропала для всех. Если специализированные сайты несут хотя бы минимальную ответственность за сохранность своего контента, то в такой сети вся ответственность ложится на плечи пользователей. Такая сеть пользователей может быть более надёжной как хранилище контента чем центральный сервер. только при условии что этих пользователей много и все они активны. 
- Большинство сайтов (особенно сайты торрентов) максимально не пригодны для програмной обработки: у большинства есть только один способ искать аниме -- по названию. Ввёл не то название, ошибся, опечатался, написал на другом языке, с другим переводом -- и аниме не будет найдено. Из специализированных, которые я видел, только у smotret-anime.online есть удобный именно для программы поиск и не по названию а по ID. 
- Торрент-сайты блокирую не меньше (если не больше) чем сайты подобные smotret-anime.online. То есть даже внедри я подобный крайне сложный механизм, нет никаких гарантий, что его не заблокируют точно так же.
